tools/payloads/ysoserial/find_ysoserial_offsets.rb

```diff
@@ -8,38 +8,46 @@
 YSOSERIAL_RANDOMIZED_HEADER = 'ysoserial/Pwner'
 PAYLOAD_TEST_MIN_LENGTH = 4
 PAYLOAD_TEST_MAX_LENGTH = 5
+YSOSERIAL_MODIFIED_TYPES = ['cmd', 'bash', 'powershell', 'none']
+YSOSERIAL_ALL_TYPES = ['original'] + YSOSERIAL_MODIFIED_TYPES
 
 # ARGV parsing
 if ARGV.include?("-h")
   puts 'ysoserial object template generator'
   puts
   puts 'Usage:'
-  puts '  -h          Help'
-  puts '  -d          Debug mode (output offset information only)'
-  puts "  -m [type]   Use 'ysoserial-modified' with the specified payload type"
+  puts '  -h             Help'
+  puts '  -d             Debug mode (output offset information only)'
+  puts "  -m [type]      Use 'ysoserial-modified' with the specified payload type"
+  puts '  -p [payloads]  Specified ysoserial payload (payloads1,payloads2,...)'
+  puts '  -a             Generate all types of payloads'
   puts
   abort
 end
 
-debug = ARGV.include?('-d')
-ysoserial_modified = ARGV.include?('-m')
-if ysoserial_modified
-  payload_type = ARGV[ARGV.find_index('-m')+1]
-  unless ['cmd', 'bash', 'powershell', 'none'].include?(payload_type)
+@generate_all = ARGV.include?('-a')
+@debug = ARGV.include?('-d')
+@ysoserial_modified = ARGV.include?('-m')
+if @ysoserial_modified
+  @payload_type = ARGV[ARGV.find_index('-m')+1]
+  unless YSOSERIAL_MODIFIED_TYPES.include?(@payload_type)
     STDERR.puts 'ERROR: Invalid payload type specified'
     abort
   end
 end
+if index = ARGV.index('-p')
```

Comment:
It looks like this is assigning a variable in the conditional, rather than variable comparison.

If so, please wrap in brackets for clarity.

```suggestion
if (index = ARGV.index('-p'))
```